File: stm-servicos/src/main/java/io/github/franciscosviana/stmservicos/domain/model/Tecnico.java
package io.github.franciscosviana.stmservicos.domain.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Tecnico {

    @Id
    private UUID id;

    @Column(unique = true)
    private Long codigo;

    private String nome;
    private String cpf;

    private String telefone;
    private String email;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "credenciado_id", nullable = false)
    private Credenciado credenciado;

    @Embedded
    private Endereco endereco;
}

File: stm-servicos-front/src/app/tecnicos/tecnico.ts
import { Endereco } from "../ordem-servico/endereco";

export class Tecnico {

    id?: string;        // UUID
    codigo?: string;    // Long

    nome?: string;
    cpf?: string;

    telefone?: string;
    email?: string;

    endereco?: Endereco;
}
